waspc/src/Wasp/Lib.hs

```diff
@@ -37,73 +40,76 @@ compile ::
   CompileOptions ->
   IO ([CompileWarning], [CompileError])
 compile waspDir outDir options = do
-  (analyzerWarnings, appSpecOrAnalyzerErrors) <- analyzeWaspProject waspDir options
+  appSpecOrAnalyzerErrors <- findAndAnalyzeWaspFile waspDir options
+  compileWarnings <- warnIfDotEnvPresent waspDir
   compilerWarningsAndErrors <- case appSpecOrAnalyzerErrors of
     Left analyzerErrors -> return ([], toList analyzerErrors)
     Right appSpec ->
       case ASV.validateAppSpec appSpec of
         [] -> do
           (generatorWarnings, generatorErrors) <- Generator.writeWebAppCode appSpec outDir (sendMessage options)
           return (map show $ generatorWarningsFilter options generatorWarnings, map show generatorErrors)
-        validationErrors -> do
-          return ([], map show validationErrors)
-  return $ (analyzerWarnings, []) <> compilerWarningsAndErrors
+        validationErrors -> return ([], map show validationErrors)
+  return $ (compileWarnings, []) <> compilerWarningsAndErrors
 
-analyzeWaspProject ::
+findAndAnalyzeWaspFile ::
   Path' Abs (Dir WaspProjectDir) ->
   CompileOptions ->
-  IO ([CompileWarning], Either (NonEmpty CompileError) AS.AppSpec)
-analyzeWaspProject waspDir options = do
-  maybeWaspFilePath <- findWaspFile waspDir
-  appSpecOrAnalyzerErrors <- case maybeWaspFilePath of
-    Nothing -> return $ Left $ fromList ["Couldn't find a single *.wasp file."]
-    Just waspFilePath -> do
-      waspFileContent <- readFile (SP.fromAbsFile waspFilePath)
-      case Analyzer.analyze waspFileContent of
-        Left analyzeError ->
-          return $
-            Left $
-              fromList
-                [ showCompilerErrorForTerminal
-                    (waspFilePath, waspFileContent)
-                    (getErrorMessageAndCtx analyzeError)
-                ]
-        Right decls -> do
-          externalCodeFiles <-
-            ExternalCode.readFiles (CompileOptions.externalCodeDirPath options)
-          maybeDotEnvServerFile <- findDotEnvServer waspDir
-          maybeDotEnvClientFile <- findDotEnvClient waspDir
-          maybeMigrationsDir <- findMigrationsDir waspDir
-          return $
-            Right
-              AS.AppSpec
-                { AS.decls = decls,
-                  AS.externalCodeFiles = externalCodeFiles,
-                  AS.externalCodeDirPath = CompileOptions.externalCodeDirPath options,
-                  AS.migrationsDir = maybeMigrationsDir,
-                  AS.dotEnvServerFile = maybeDotEnvServerFile,
-                  AS.dotEnvClientFile = maybeDotEnvClientFile,
-                  AS.isBuild = CompileOptions.isBuild options
-                }
-  analyzerWarnings <- warnIfDotEnvPresent waspDir
-  return (analyzerWarnings, appSpecOrAnalyzerErrors)
+  IO (Either (NonEmpty CompileError) AppSpec)
+findAndAnalyzeWaspFile waspDir options = runExceptT $ do
```

Comment:
Good attempt with the new name, but I don't believe it is a correct name -> we also read external code files, looks for migration dir, look for env files, ... .
That is why it made sense to say `analyzeWaspProject`, because it takes whole project into account, analyzes it, and produces AppSpec. It is not result of analysing just a single Wasp file, but more things. Also, in the future we expect this to work with multiple Wasp files. So I would just leave the old name.